File: file-templates/templates/MVVM/TableList.xctemplate/UIViewObjective-C/___FILEBASENAME___Cell.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"
#import "___VARIABLE_productName___CellModel.h"
#import <Masonry/Masonry.h>

@interface ___VARIABLE_productName___Cell ()
/// 容器视图
@property (nonatomic, strong) UIView *containerView;
@end

@implementation ___VARIABLE_productName___Cell

#pragma mark - private

- (void)onUpdateUIComponents {
    
}

- (void)onDataBindings {
    [self onUpdateUIComponents];
}

#pragma mark - setter

- (void)setCellModel:(___VARIABLE_productName___CellModel *)cellModel {
    if (_cellModel != cellModel) {
        _cellModel = cellModel;
        [self onDataBindings];
    }
}

- (UIView *)containerView {
    if (!_containerView) {
        UIView *view = [[UIView alloc] initWithFrame:CGRectZero];
        [self.contentView addSubview:view];
        [view mas_makeConstraints:^(MASConstraintMaker *make) {
            make.top.equalTo(self.contentView.mas_top).offset(0);
            make.left.equalTo(self.contentView.mas_left).offset(0);
            make.right.equalTo(self.contentView.mas_right).offset(0);
            make.bottom.equalTo(self.contentView.mas_bottom).offset(0);
        }];
        _containerView = view;
    }
    return _containerView;
}

@end

File: file-templates/templates/MVVM/TableList.xctemplate/UIViewControllerSwift/___FILEBASENAME___Cell.swift
//___FILEHEADER___

import UIKit
import SnapKit

class ___VARIABLE_productName___Cell: UITableViewCell {
    
    public var provider: ___VARIABLE_productName___CellModelProvider? {
        didSet {
            setupBindings()
        }
    }
    
    /// 容器视图
    private lazy var containerView: UIView = {
        let view = UIView()
        view.backgroundColor = .white
        return view
    }()
    
    // MARK: - UIComponents
    
    public func addUIComponents() {
        contentView.addSubview(containerView)
    }
    
    public func layoutUIComponents() {
        containerView.snp.makeConstraints { make in
            make.edges.equalTo(UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0))
        }
    }

    public func setupBindings() {
        onUpdateUIComponents()
    }
    
    public func onUpdateUIComponents() {
        guard let provider = self.provider else { return }
        // Configure components for provider
    }

}
